DataStructures/SegmentTrees/SegmentTreeApply.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+
+namespace DataStructures.SegmentTrees
+{
+    /// <summary>
+    /// This is an extension of a segment tree, which allows to apply distributive operations to a subarray (in this case muliplication).
+    /// </summary>
+    public class SegmentTreeApply : SegmentTree
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SegmentTreeApply"/> class.
+        /// Runtime complexity: O(n) where n equals the array-length.
+        /// </summary>
+        /// <param name="arr">Array on which the operations should be made.</param>
+        public SegmentTreeApply(int[] arr)
+        : base(arr)
+        {
+            // Initilizes and fills "operation" array with neutral element (in this case 1, because value * 1 = value)
+            Operation = new int[Tree.Length];
+            Array.Fill<int>(Operation, 1);
+        }
+
+        /// <summary>
+        /// Gets an array that stores for each node an operation,
+        /// which must be applied to all direct and indirect child nodes of this node
+        /// (but not to the node itself).
+        /// </summary>
+        public int[] Operation { get; }
```

Comment:
It's not an operation, it's more like operand? Stuff other stuff gets multiplied by? Need to improve naming here